packages/notification-center-vue/src/lib/notification-center-component/NotificationCenterComponent.vue

```diff
@@ -1,53 +1,228 @@
+
+    return {
+      popoverArrowClass,
+      popoverDropdownClass,
+      bellButtonClass,
+      gradientDotClass,
+      bellColors,
+      unseenCount: undefined as number | undefined,
+      hasSlot: !!this.$slots.default,
+    };
+  },
+  watch: {
+    theme: {
+      handler(val) {
+        const { popoverArrowClass, popoverDropdownClass, bellButtonClass, gradientDotClass, bellColors } =
+          this.calculateStyles({ theme: val, styles: this.styles, colorScheme: this.colorScheme });
+        this.popoverArrowClass = popoverArrowClass;
+        this.popoverDropdownClass = popoverDropdownClass;
+        this.bellButtonClass = bellButtonClass;
+        this.gradientDotClass = gradientDotClass;
+        this.bellColors = bellColors;
+      },
+      deep: true,
+    },
+    styles: {
+      handler(val) {
+        const { popoverArrowClass, popoverDropdownClass, bellButtonClass, gradientDotClass, bellColors } =
+          this.calculateStyles({ theme: this.theme, styles: val, colorScheme: this.colorScheme });
+        this.popoverArrowClass = popoverArrowClass;
+        this.popoverDropdownClass = popoverDropdownClass;
+        this.bellButtonClass = bellButtonClass;
+        this.gradientDotClass = gradientDotClass;
+        this.bellColors = bellColors;
+      },
+      deep: true,
+    },
+    colorScheme: {
+      handler(val) {
+        const { popoverArrowClass, popoverDropdownClass, bellButtonClass, gradientDotClass, bellColors } =
+          this.calculateStyles({ theme: this.theme, styles: this.styles, colorScheme: val });
+        this.popoverArrowClass = popoverArrowClass;
+        this.popoverDropdownClass = popoverDropdownClass;
+        this.bellButtonClass = bellButtonClass;
+        this.gradientDotClass = gradientDotClass;
+        this.bellColors = bellColors;
+      },
+    },
+  },
+  mounted() {
+    const arrowsContainer = (this.$refs.popover as any).$refs.popper.$_arrowNode as HTMLDivElement;
+    arrowsContainer.childNodes.forEach((node) => {
+      (node as HTMLDivElement).classList.add(this.popoverArrowClass);
+    });
+
+    // listen to the unseen count changed event propagated from the web component
+    document.addEventListener('novu:unseen_count_changed', (event) => {
+      this.unseenCount = (event as CustomEvent).detail as number;
+    });
+  },
+  methods: {
+    calculateStyles: ({
+      styles,
+      theme: propsTheme,
+      colorScheme,
+    }: Pick<NotificationCenterComponentProps, 'theme' | 'styles' | 'colorScheme'>) => {
+      const { theme, common } = getDefaultTheme({ colorScheme, theme: propsTheme });
+      const { bellColors } = getDefaultBellColors({
+        colorScheme,
+        bellColors: {},
+      });
+
+      return {
+        popoverArrowClass: css(
+          getStyleByPath({
+            styles,
+            path: 'popover.arrow',
+            theme,
+            common,
+            colorScheme,
+          })
```

Comment:
`getStyleByPath` is the function that takes the `style` part object from the `styles` prop by a specific `path`. Then it's used to create CSS class.